src/Microsoft.OData.Client/DataServiceContext.cs

```diff
@@ -932,25 +934,25 @@ public void CancelRequest(IAsyncResult asyncResult)
             BaseAsyncResult result = asyncResult as BaseAsyncResult;
 
             // verify this asyncResult orginated from this context or via query from this context
-            if ((null == result) || (this != result.Source))
+            if ((result == null) || (this != result.Source))
             {
                 object context = null;
                 DataServiceQuery query = null;
-                if (null != result)
+                if (result != null)
                 {
                     query = result.Source as DataServiceQuery;
 
-                    if (null != query)
+                    if (query != null)
                     {
                         DataServiceQueryProvider provider = query.Provider as DataServiceQueryProvider;
-                        if (null != provider)
+                        if (provider != null)
                         {
                             context = provider.Context;
                         }
                     }
                 }
 
-                if (this != context)
+                if (context != null)

```

Comment:
@xuzhg `if (this != context)` changed to `if (context != null)`. Kindly verify this change...

Reply:
@gathogojr Thanks, it's a typo. I fixed that with a force push.